custom/icds_reports/tasks.py

```diff
@@ -1209,6 +1214,40 @@ def build_incentive_report(agg_date=None):
                 build_incentive_files.delay(location, agg_date, file_format, 3, location.parent.parent, location.parent)
 
 
+def is_launched_awc_incentive_report(launched_awc_ids):
+    awc_ids = launched_awc_ids.values_list('awc_id', flat=True)
+    awc_launched_count_from_aggregate = AggAwc.objects.filter(aggregation_level=5, awc_id__in=awc_ids).all()
```

Comment:
These two tables should contain identical lists of awcs, so the filter is not needed and matching a list of 600000 will make the query significantly slower. Also the `.all()` is unnecessary with the django orm. You probably only want to pull awc_id and is_launched here as well (the two columns used in the actual data access below). Same comment as above about incremental building, it is hard to follow